compiler/resolution/preFold.cpp

```diff
@@ -1520,11 +1520,26 @@ static bool isInstantiatedField(Symbol* field) {
   AggregateType* at     = toAggregateType(ts->type);
   bool           retval = false;
 
-  for_formals(formal, at->typeConstructor) {
-    if (strcmp(field->name, formal->name) == 0) {
-      if (formal->hasFlag(FLAG_TYPE_VARIABLE) == true) {
-        retval = true;
-        break;
+  // BHARSH INIT TODO: Sometimes the type constructor is not resolved when
+  // initializers are used. Why?
```

Comment:
I believe we intentionally skip them when the type is concrete, or when we are working on an instantiation (and thus the type constructor on the generic base is the only one resolved), but that information might be out of date